rd_server.py

```diff
@@ -505,6 +505,28 @@ def find_image_work(items, default = None):
             return work
     return default
 
+def find_multislot_work(items, default = None):
+    for work in items:
+        # Work is of the class RDWork
+        if work.multislots:
+            return work
+    return default
+
+def check_multislot_work(work):
+    return work.multislots
+
+def free_a_slot(this_free_slots, current_slot_host, current_slot_id):
+    for slot_iterator in range(len(free_slots)-1, -1,-1):
+        target_slot = this_free_slots[slot_iterator]
+        target_slot_host = target_slot.machine.host
+        target_slot_id = target_slot.work_root
+        if current_slot_host == target_slot_host:
+            if current_slot_id != target_slot_id:
+                print("Found a slot to be removed;target;", target_slot, target_slot_host, target_slot_id, "current", current_slot_host, current_slot_id)
+                this_free_slots.remove(target_slot)
+                break
+    return this_free_slots
+
 def scheduler_tick():
```

Comment:
TODO: Signal multi-slot (`slot2`)in the `slot` and explicitly free the slot on completion, 

https://github.com/xiph/rd_tool/pull/151/files#diff-06f1b93a0b586bcde49cd855f722066de9daaedbf01393b0b90b6670b1ba2109R569